### ISSUE TITLE
[ApiView] Support adding comments to removed lines

### ISSUE THREAD

**Issue by alzimmermsft** [Azure/azure-sdk-tools#8420]
At the time of writing this issue, ApiView doesn't support adding comments to removed lines of code in a diff. It's a valid case to do so if there are questions on why the API was removed (as this is a major breaking change).

It'd be nice to support this rather than the commenter needing to add the comment to an added or unchanged line and saying the comment is actually for a different line in the ApiView diff.

**Comment by swathipil** [comment]
Hi @chidozieononiwu , is there an ETA on this? We'd also like support for the reasons listed above.

**Comment by maririos** [comment]
We’ve decided to prioritize APIView server changes to accelerate Copilot work, so the UI feature won’t happen unless it’s blocking users or directly supports Copilot. How critical is it for you, and is there any workaround if we postpone it?

**Comment by alzimmermsft** [comment]
I believe this may be important to the Copilot work as how will it be able to put comments on lines of code removed if it determines it needs to.

**Comment by maririos** [comment]
> I believe this may be important to the Copilot work as how will it be able to put comments on lines of code removed if it determines it needs to.

@tjprescott what do you think?

**Comment by tjprescott** [comment]
The use case is just as legitimate for copilot as it would be for human architects who might want to comment on removals.

The challenging piece is just how the commenting system works under-the-hood, but that's no reason not to investigate. 

@AlitzelMendez could you set aside some time to investigate the technical feasibility of this, assuming that the removed lines in fact had a legitimate line ID.

**Comment by AlitzelMendez** [comment]
@tjprescott @maririos I took some time to investigate this issue.

Let me explain how this works under the hood. When comparing two code files, the result is a merged view containing lines from both files, where:
- Removed lines are marked as removed (-)
- Added lines are marked as added (+)
- Unchanged lines remain unmarked

**Example:**

**File A:**
```
Hello - id H
World - id W
! - id Z
```

**File B:**
```
Hello - id H
Everyone - id E
! - id Z
``` 

**File A vs File B (Diff View):**
```
Hello - id H
(-) World - id W
(+) Everyone - id E
! - id Z
```

**Key Point:** Removed lines retain their legitimate line ID from the original revision, which means it is technically feasible to add comments to removed lines.

**Implications:**

Since removed lines have legitimate IDs belonging to their original revision:
- Comments on removed lines will still appear in the revision where the line existed
- These comments will also be visible in the Conversations view, which can make it challenging to locate what they're referring to since they point to an old revision
- However, in the diff view, the comments remain consistent and easy to understand

**Comment by maririos** [comment]
>Comments on removed lines will still appear in the revision where the line existed

To clarify:
- If someone added a comment on a previous revision on that specific line, the comment would appear in the new revision on the removed line
- If someone wants to add a comment in a removed line, they should:
  - Go to the revision that had the line and comment there
  - Comment on the current revision on the closes line present in current revision

**Comment by tjprescott** [comment]
What I understand is that if we made it so you could comment on the "removed line" in diff view, it would appear there, no problem, but if you flipped back to a rev where that line was NOT removed, your comment about it being removed would still appear, which would be confusing.

**Comment by tjprescott** [comment]
I'm making some assumptions about how comment filtering here, but let's take your example:

**Rev A**

```
1: Hello - id H
2: World - id W
3: ! - id Z
```

**Rev B**

```
Hello - id H
Everyone - id E
! - id Z
```

**Diff of B to A**

```
1: Hello - id H
2: (-) World - id W
3: (+) Everyone - id E
4: ! - id Z
```

So, if in this Diff, I make a comment on line 2, perhaps we could save the comment with the removed line ID but associated with Rev B rather than A. 

Thus, when on Rev A, the comment would not appear since it didn't exist until Rev B. This would ensure it appears in the diff view, but not when looking at Rev A.

**Comment by AlitzelMendez** [comment]
> What I understand is that if we made it so you could comment on the "removed line" in diff view, it would appear there, no problem, but if you flipped back to a rev where that line was NOT removed, your comment about it being removed would still appear, which would be confusing.

exactly this! 

> So, if in this Diff, I make a comment on line 2, perhaps we could save the comment with the removed line ID but associated with Rev B rather than A.

I think this is an interesting option! I like it! Although it would require some significant changes. Right now, we show comments independently of the revision: we retrieve all comments for the review, and if there is a comment for that element ID, it is displayed (which is why we see comments for very old revisions). In other conversations, we’ve discussed the actual benefit of doing this—showing comments for other revisions.

**Comment by AlitzelMendez** [comment]
> > Comments on removed lines will still appear in the revision where the line existed
> 
> To clarify:
> 
> * If someone added a comment on a previous revision on that specific line, the comment would appear in the new revision on the removed line
> * If someone wants to add a comment in a removed line, they should:
>   
>   * Go to the revision that had the line and comment there
>   * Comment on the current revision on the closes line present in current revision

right now is not happening because comments are disable for removed lines, it is just the implications that it will have if we enable the feature

**Comment by maririos** [comment]
Thank you for the detailed explanation. I have added this issue into the `comments` epic so it is part of the overall comments flow conversation

**Comment by tjprescott** [comment]
Yes, I think this is definitely doable and I think filtering comments by revision is the biggest change, but a logical one in general that would resolve the potential UX issues with this. That being said, I would also not consider it a high priority item.

**Comment by tjprescott** [comment]
@maririos isn't this basically a duplicate of https://github.com/Azure/azure-sdk-tools/issues/2506? If so, I would recommend we keep this and close that one as a duplicate in favor of this.
